File: frontend/src/api/productApi.ts
// src/api/productApi.ts
import axiosClient from './axiosClient.ts';

// Có thể define type riêng cho product
import { Category } from '../types/Category.ts';
import type { Product } from '../types/Product';

export const productApi = {
  // Limit mặc định là 6, page mặc định là 1, limit là số sản phẩm tối đa mỗi trang
  getProducts: async (page: number = 1, limit: number = 6, category?: string, keyword: string = "") => {
    const params: any = { page, limit, keyword };
    if (category) params.category = category;

    const res = await axiosClient.get('/products', { params });
    console.log("📦 API response:", res);
    return res;
  },


  getProductById: async (id: string): Promise<{data: Product}> => {
    return await axiosClient.get(`/products/${id}`);
  },

//   getSimilarProducts: async (id: string) => {
//     return await axiosClient.get(`/products/${id}/similar`);
//   },

//   getNewestProducts: async () => {
//     return await axiosClient.get('/newest');
//   },

//   getBestSellers: async () => {
//     return await axiosClient.get('/best-sellers');
//   },

//   createProduct: async (product: Partial<Product>) => {
//     return await axiosClient.post('/create-products', product);
//   }
};

File: Mobileptit/app/src/main/java/com/ptitshop/app/data/remote/api/ProductApi.java
package com.ptitshop.app.data.remote.api;

import com.ptitshop.app.data.model.Product;
import com.ptitshop.app.data.remote.dto.response.*;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.*;

public interface ProductApi {
    @GET("v1/api/products")
    Call<ProductListResponse> getProducts(@Query("page") int page, @Query("limit") int limit, @Query("category") String category, @Query("keyword") String keyword);

    @GET("v1/api/products/{id}")
    Call<ProductDetailResponse> getProductDetail(@Path("id") String id);

    @GET("v1/api/products/{id}/similar")
    Call<List<Product>> getSimilarProducts(@Path("id") String id);

    @GET("v1/api/products/top-viewed")
    Call<List<Product>> getTopViewedProducts();

    @GET("v1/api/products/top-discount")
    Call<List<Product>> getTopDiscountProducts();

    @GET("v1/api/newest")
    Call<List<Product>> getNewestProducts();

    @GET("v1/api/best-sellers")
    Call<List<Product>> getBestSellers(@Query("limit") int limit);
}
